handlers/user.go

```diff
@@ -1298,177 +1335,193 @@ func UserUpdateEmailPost(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	session.SetValue("subscription", "")
-	session.SetValue("authenticated", false)
-	session.DeleteValue("user_id")
-
-	err = purgeAllSessionsForUser(r.Context(), uint64(user.ID))
-	if err != nil {
-		logger.Errorf("error: purging sessions for user %v: %v", user.ID, err)
-		utils.SetFlash(w, r, authSessionName, "Error: Could not Update Email.")
-		http.Redirect(w, r, "/confirmation", http.StatusSeeOther)
+	if emailExists {
+		utils.SetFlash(w, r, authSessionName, "Error: Could not update email. The new email already exists, please send a request with a different email.")
+		http.Redirect(w, r, "/user/settings", http.StatusSeeOther)
 		return
 	}
 
-	utils.SetFlash(w, r, authSessionName, "Your email has been updated successfully! <br> You can log in with your new email.")
-	http.Redirect(w, r, "/login", http.StatusSeeOther)
-}
-
-func sendEmailUpdateConfirmation(userId uint64, newEmail string) error {
-	now := time.Now()
-	emailConfirmationHash := utils.RandomString(40)
-
-	tx, err := db.FrontendWriterDB.Beginx()
-	if err != nil {
-		return err
-	}
-	defer tx.Rollback()
-
-	var lastTs *time.Time
-	err = tx.Get(&lastTs, "SELECT email_confirmation_ts FROM users WHERE id = $1", userId)
+	// update users email in DB and set stripe email pending flag
+	_, err = db.FrontendWriterDB.Exec(`UPDATE users SET email = $1, email_confirmation_hash = NULL, email_change_to_value = NULL, stripe_email_pending = $2 WHERE id = $3`, user.NewEmail, user.StripeCustomerId != "", user.ID)
 	if err != nil {
-		return fmt.Errorf("error getting confirmation-ts: %w", err)
-	}
-	if lastTs != nil && (*lastTs).Add(authConfirmEmailRateLimit).After(now) {
-		return &types.RateLimitError{TimeLeft: (*lastTs).Add(authConfirmEmailRateLimit).Sub(now)}
-	}
-
-	_, err = tx.Exec("UPDATE users SET email_confirmation_hash = $1, email_change_to_value = $2 WHERE id = $3", emailConfirmationHash, newEmail, userId)
-	if err != nil {
-		return fmt.Errorf("error updating confirmation-hash: %w", err)
-	}
-
-	err = tx.Commit()
-	if err != nil {
-		return fmt.Errorf("error committing db-tx: %w", err)
+		utils.LogError(err, "error updating email for user", 0, map[string]interface{}{"userID": user.ID, "newEmail": user.NewEmail})
+		utils.SetFlash(w, r, authSessionName, "Error: Could not update email. Please try again. If this error persists please contact <a href=\"https://support.bitfly.at/support/home\">support</a>.")
+		http.Redirect(w, r, "/user/settings", http.StatusSeeOther)
+		return
 	}
 
-	subject := fmt.Sprintf("%s: Verify your email-address", utils.Config.Frontend.SiteDomain)
-	msg := fmt.Sprintf(`To update your email on %[1]s please verify it by clicking this link:
-
-https://%[1]s/settings/email/%[2]s
-
-Best regards,
-
-%[1]s
-`, utils.Config.Frontend.SiteDomain, emailConfirmationHash)
-	err = mail.SendTextMail(newEmail, subject, msg, []types.EmailAttachment{})
-	if err != nil {
-		return err
+	if sessionUser.UserID == uint64(user.ID) {
```

Comment:
I might be wrong about that but this seems redundant. `purgeAllSessionsForUser` destroys the session store entries for the user so this special case shouldn't be necessary, right?

Reply:
seems so, fixed, ptal